block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockMetadataStore.scala

```diff
@@ -93,6 +111,12 @@ object BlockMetadataStore {
     def heightMap: F[SortedMap[Long, Set[BlockHash]]] =
       dagState.get.map(_.heightMap)
 
+    def lastFinalizedBlock(implicit sync: Sync[F]): F[BlockHash] = {
+      val errMsg =
+        "DagState does not contain lastFinalizedBlock. Are you calling this on empty BlockDagStorage? Otherwise there is a bug."
+      dagState.get.flatMap(_.lastFinalizedBlock.liftTo[F](new Exception(errMsg)))
```

Comment:
this is public API of BlockMetadataStore, it forces that when this function is called, BlockDagStorage should already contain "approved block | genesis | LFB". No one should call this before there is some meaningful DAG present in DagStorage.